### DISCUSSION TITLE
How I enable support for JS files that have JSX syntax insiede?

### DISCUSSION THREAD

**Discussion by rijans** [vitejs/vite-plugin-react-swc#91] [Ideas]
With `vite-plugin-react-swc` I get error of "_Unexpected token_" inside the start of jsx syntax. My component files are with **.js** extension, which is a very old project.

How do I enable this plugin to work with these files?

**Comment by ArnaudBarre** [comment]
Using JSX inside JS will not be supported by this plugin.
Rename your js files containing jsx to .jsx or .tsx.
More info: https://github.com/vitejs/vite-plugin-react-swc/issues/57#issuecomment-1425931254
